backend/src/services/notifications.ts

```diff
@@ -0,0 +1,132 @@
+import * as admin from "firebase-admin";
+import schedule from "node-schedule";
+import { Mentee, Mentor, Session } from "../models";
+import { InternalError } from "../errors";
+
+/**
+ * Unicode notes:
+ * eyes - 1F440
+ * cool with glasses - 1F60E
+ * check mark - 2705
+ * broken heart - 1F494
+ *
+ * To find emojis and their unicode value, use https://apps.timwhitlock.info/emoji/tables/unicode
+ * The unicode is in format "U+1234"; in code, use "\u{1234}"
+ */
+
+async function sendNotification(title: string, body: string, deviceToken: string) {
+  const message = {
+    notification: {
+      title,
+      body,
+    },
+    token: deviceToken,
+  };
+
+  admin
+    .messaging()
+    .send(message)
+    .then((response) => {
+      console.log("Notification sent successfully:", response);
+    })
+    .catch((error) => {
+      console.error("Error sending notification:", error);
+    });
+}
+
+async function startUpcomingSessionCronJob() {
+  schedule.scheduleJob("*/1 * * * *", async () => {
+    try {
+      const upcomingNotifSessions = await Session.find({
+        upcomingSessionNotifSent: { $eq: false },
+      });
+      upcomingNotifSessions.forEach(async (session) => {
+        const dateNow = new Date();
+        const mentee = await Mentee.findById(session.menteeId);
+        if (!mentee) {
+          throw InternalError.ERROR_GETTING_MENTEE;
+        }
+        const mentor = await Mentor.findById(session.mentorId);
+        if (!mentor) {
+          throw InternalError.ERROR_GETTING_MENTOR;
+        }
+        if (session.startTime.getTime() - dateNow.getTime() <= 86400000) {
+          if (session.preSessionCompleted) {
+            const menteeNotif = await sendNotification(
+              "You have an upcoming session.",
+              `Ready for your session with ${mentor.name} in 24 hours? \u{1F440}`,
```

Comment:
Since the only difference between the if statement is the mentee and mentor message, the if-else statement should modify a variable for the message to be sent based on the pre-session notes. Then after the if-else statement send the notification. This will help clean up the code and reduce repitition.